ocp_resources/datavolume.py

```diff
@@ -216,20 +216,85 @@ def scratch_pvc(self):
             client=self.client,
         )
 
+    def _dv_phase(self):
+        if self.exists:
+            return self.instance.status.phase
+        return DV_DELETED_BY_GC
+
     def _check_none_pending_status(self, failure_timeout=120):
         # Avoid waiting for "Succeeded" status if DV's in Pending/None status
         sample = None
         try:
             for sample in TimeoutSampler(
                 wait_timeout=failure_timeout,
                 sleep=10,
-                func=lambda: self.instance.status.phase,
+                func=self._dv_phase,
             ):
                 # If DV status is Pending (or Status is not yet updated) continue to wait, else exit the wait loop
                 if sample in [self.Status.PENDING, None]:
                     continue
-                else:
-                    break
+                break
         except TimeoutExpiredError:
-            LOGGER.error(f"{self.name} status is {sample}")
+            self.logger.error(f"{self.name} status is {sample}")
             raise
+
+    def is_garbage_collector_enabled_dv(self):
+        if self.exists:
+            dv_annotations = self.instance.metadata.get("annotations")
+            if dv_annotations:
+                return (
+                    dv_annotations.get(
+                        f"{Resource.ApiGroup.CDI_KUBEVIRT_IO}/storage.deleteAfterCompletion"
+                    )
+                    is not False
+                )
+            return True
+        return (
+            self.pvc.exists and self.pvc.instance.status.phase == self.pvc.Status.BOUND
+        )
```

Comment:
the function is `is_garbage_collector_enabled_dv` - this return True due to a bug where GC is disabled, dv was deleted, pcv exists and boud

Reply:
if the dv already deleted that is the the indication to the dv gc, if we had this beg as you described it really a serious problem of resource cleanup

Reply:
how will you catch it?

Reply:
Arnon will add it to tier1 test with the gc disabled